Short names in export graphviz file

## Hello!
<!-- Please leave this section as-is, it's designed to help others in the community know how to interact with our GitHub issues. -->

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

When I export a dependency graph to a file for visualization, the nodes have very long names, based on their URNs. This makes them easy to identify, but makes the resulting graphs very hard to read since the node labels are very long.

It would be nice if there was an option that would let me use "short" names (only the last part of the URN, the part which contains the resource name as defined in the program) when generating the graph file.

I would prefer it if node names were short and simple; instead of "urn:pulumi:my_very_long_stack_name::my_very_long_project_name::gcp:secretmanager/secretVersion:SecretVersion::some-secret-version", I would rather just see "some-secret-version". 

### Affected area/feature

CLI

@iwahbe I've done a draft PR. And as per your suggestion, I tried doing make build but I'm getting the following errors : 
```BUILD:
Protobuffer interfaces are ....... up to date
Checking for go . ✓
cd pkg && go install -ldflags "-X github.com/pulumi/pulumi/pkg/v3/version.Version=3.46.2-dev.0" github.com/pulumi/pulumi/pkg/v3/cmd/pulumi
# runtime/internal/sys
/home/gitpod/go/src/runtime/internal/sys/consts.go:13:7: StackGuardMultiplier redeclared in this block
        /home/gitpod/go/src/runtime/internal/sys/arch.go:27:7: other declaration of StackGuardMultiplier
/home/gitpod/go/src/runtime/internal/sys/consts.go:16:7: DefaultPhysPageSize redeclared in this block
        /home/gitpod/go/src/runtime/internal/sys/arch.go:36:7: other declaration of DefaultPhysPageSize
/home/gitpod/go/src/runtime/internal/sys/consts.go:20:7: PCQuantum redeclared in this block
        /home/gitpod/go/src/runtime/internal/sys/arch.go:40:7: other declaration of PCQuantum
/home/gitpod/go/src/runtime/internal/sys/consts.go:23:7: Int64Align redeclared in this block
        /home/gitpod/go/src/runtime/internal/sys/arch.go:43:7: other declaration of Int64Align
/home/gitpod/go/src/runtime/internal/sys/consts.go:30:7: MinFrameSize redeclared in this block
        /home/gitpod/go/src/runtime/internal/sys/arch.go:50:7: other declaration of MinFrameSize
/home/gitpod/go/src/runtime/internal/sys/consts.go:34:7: StackAlign redeclared in this block
        /home/gitpod/go/src/runtime/internal/sys/arch.go:54:7: other declaration of StackAlign
make: *** [Makefile:64: build] Error 2```

--- a/changelog/pending/20221116--cli--adds-a-flag-that-allows-user-to-set-the-node-label-as-the-resource-name-instead-of-the-full-urn-in-the-stack-graph.yaml
+++ b/changelog/pending/20221116--cli--adds-a-flag-that-allows-user-to-set-the-node-label-as-the-resource-name-instead-of-the-full-urn-in-the-stack-graph.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: feat
+  scope: cli
+  description: Adds a flag that allows user to set the node label as the resource name instead of full URN in the stack graph

--- a/pkg/cmd/pulumi/stack_graph.go
+++ b/pkg/cmd/pulumi/stack_graph.go
@@ -40,6 +40,9 @@ var dependencyEdgeColor string
 // The color of parent edges in the graph. Defaults to #AA6639, an orange.
 var parentEdgeColor string
 
+// Whether or not to return resource name as the node label for each node of the graph.
+var shortNodeName bool
+
 func newStackGraphCmd() *cobra.Command {
 	var stackName string
 
@@ -98,6 +101,8 @@ func newStackGraphCmd() *cobra.Command {
 		"Sets the color of dependency edges in the graph")
 	cmd.PersistentFlags().StringVar(&parentEdgeColor, "parent-edge-color", "#AA6639",
 		"Sets the color of parent edges in the graph")
+	cmd.PersistentFlags().BoolVar(&shortNodeName, "short-node-name", false,
+		"Sets the resource name as the node label for each node of the graph")
 	return cmd
 }
 
@@ -178,6 +183,9 @@ func (vertex *dependencyVertex) Data() interface{} {
 }
 
 func (vertex *dependencyVertex) Label() string {
+	if shortNodeName {
+		return string(vertex.resource.URN.Name())
+	}
 	return string(vertex.resource.URN)
 }
 

